libs/application/core/src/lib/ApplicationTemplateHelper.ts

```diff
@@ -58,8 +77,34 @@ export class ApplicationTemplateHelper<
    * @param event A state machine event
    * returns [hasChanged, newState, newApplication] where newApplication has the updated state value
    */
-  changeState(event: Event<TEvents>): [boolean, string, Application] {
-    this.initializeStateMachine()
+  changeState(
+    event: Event<TEvents>,
+    sendEmail: (emailTemplate: SendMailOptions) => Promise<string>,
```

Comment:
could this be undefined and we just provide default behavior (do nothing?) when this is undefined?

Reply:
yes, I'll fix that